extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Utils.java

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.s3;
+
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.profiles.ProfileFileLocation;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+class S3Utils {
+
+    /**
+     * Aggregates the profile files for configuration and credentials files into a single {@link ProfileFile}.
+     *
+     * @param configFilePath An {@link Optional} containing the path to the configuration file.
+     * @param credentialsFilePath An {@link Optional} containing the path to the credentials file.
+     *
+     * @return A {@link ProfileFile} that aggregates the configuration and credentials files.
+     */
+    static ProfileFile aggregateProfileFile(
+            @NotNull Optional<Path> configFilePath,
+            @NotNull Optional<Path> credentialsFilePath) {
+        final ProfileFile.Aggregator builder = ProfileFile.aggregator();
+
+        // Add the configuration file
+        configFilePath = configFilePath.or(ProfileFileLocation::configurationFileLocation);
+        configFilePath.ifPresent(path -> addProfileFile(builder, ProfileFile.Type.CONFIGURATION, path));
+
+        // Add the credentials file
+        credentialsFilePath = credentialsFilePath.or(ProfileFileLocation::credentialsFileLocation);
+        credentialsFilePath.ifPresent(path -> addProfileFile(builder, ProfileFile.Type.CREDENTIALS, path));
```

Comment:
I don't know if order matters; but `software.amazon.awssdk.profiles.ProfileFile#defaultProfileFile` applies the credentials file first, and then the config file, and then uses a LinkedHashMap to build the results. To be safe, I think we should add them in the same order.